server/memphis_handlers_poison_messages.go

```diff
@@ -28,7 +28,7 @@ import (
 
 type PoisonMessagesHandler struct{ S *Server }
 
-func (pmh PoisonMessagesHandler) HandleNewMessage(_ string, msg []byte) {
+func (pmh PoisonMessagesHandler) HandleNewMessage(_ *client, _, _ string, msg []byte) {
```

Comment:
this function is getting called from a subject subscriber, sits inside the broker so which client is passing here?

Reply:
the subscription gives the message handler all the details of the arriving message, including the client from which it came from, this is the client in this case. checked and verified :)